src/parse-nm.c

```diff
@@ -225,6 +225,11 @@ parse_routes(GKeyFile* kf, const gchar* group, GArray** routes_arr)
         /* Append gateway/via IP */
         if (split[0] && split[1])
             route->via = g_strdup(split[1]); //no need to free, will stay in netdef
+        else {
+            /* NM assumes a "default" route if this is not defined in the keyfile.
+             * See nm-settings-ip-config.c -> nm_ip_route_set_next_hop()/canonicalize_ip(). */
+            route->via = g_strdup("default");
```

Comment:
As mentioned in a private chat, I have doubts regarding this. The concept of "default" to designate a gateway doesn't make sense to me, and we suspect this is actually used to specify local routes